tensorflow_addons/seq2seq/decoder.py

```diff
@@ -302,26 +302,19 @@ def dynamic_decode(
     Raises:
       ValueError: if `maximum_iterations` is provided but is not a scalar.
     """
-    with tf.compat.v1.variable_scope(scope, "decoder") as varscope:
-        # Determine context types.
-        ctxt = tf.compat.v1.get_default_graph()._get_control_flow_context()
-        is_xla = control_flow_util.GetContainingXLAContext(ctxt) is not None
-        in_while_loop = control_flow_util.GetContainingWhileContext(ctxt) is not None
-        # Properly cache variable values inside the while_loop.
-        # Don't set a caching device when running in a loop, since it is
-        # possible that train steps could be wrapped in a tf.while_loop. In that
-        # scenario caching prevents forward computations in loop iterations from
-        # re-reading the updated weights.
-        if not tf.executing_eagerly() and not in_while_loop:
-            if varscope.caching_device is None:
```

Comment:
The caching device will have a big performance boost if the code is run in a remote parameter server and worker setting, so that the variable is not read for every iteration of while loop. Is there any reason to remove it?

Reply:
Ah, you are right. The code is only expect to work in tf v1. Removing it for v2 should be fine.